app/models/user.rb

```diff
@@ -1478,8 +1478,14 @@ def trigger_user_updated_event
 
   def check_if_title_is_badged_granted
     if title_changed? && !new_record? && user_profile
-      badge_granted_title = title.present? && badges.where(allow_title: true, name: title).exists?
-      user_profile.update_column(:badge_granted_title, badge_granted_title)
+      badge_matching_title = title.present? && badges.find do |badge|
+        badge.allow_title? && (badge.display_name == title || badge.name == title)
+      end
+      user_has_badge_like_title = badge_matching_title.present?
+      user_profile.update(
+        badge_granted_title: user_has_badge_like_title,
+        granted_title_badge_id: user_has_badge_like_title ? badge_matching_title.id : nil
```

Comment:
Super minor but you can do `badge_matching_title&.id` here instead of inline if.

Reply:
No problem. We had banned the safe navigation operator at my old job so I need to get myself in the habit of using it